pkg/segment/query/metadata/segmentmicroindex.go

```diff
@@ -99,7 +99,12 @@ func (smi *SegmentMicroIndices) clearMicroIndices() {
 
 // Returns all columnar cmis for a given block or any errors encountered
 func (smi *SegmentMicroIndices) GetCMIsForBlock(blkNum uint16) (map[string]*structs.CmiContainer, error) {
-	if int(blkNum) >= len(smi.blockCmis) {
+	if len(smi.blockCmis) == 0 {
+		log.Errorf("GetCMIsForBlock: No block cmis are loaded. loadedMicroIndices: %+v", smi.loadedMicroIndices)
+		return nil, fmt.Errorf("no cmis are loaded")
+	}
+
+	if int(blkNum) >= len(smi.blockCmis)-1 {
```

Comment:
our `blkNum` start with 0, so when len(smi.blockCmis) is 1, so even though 1 block has been loaded, we will incorrectly return error here ?

Reply:
Yes, I will remove this.